examples/shell/cmd_device.cpp

```diff
@@ -0,0 +1,374 @@
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "DeviceId:        ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetDeviceId(value64));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(&value64), sizeof(value64));
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetDeviceCert(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint8_t buf[512];
+    size_t bufSize;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "DeviceCert:      ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetDeviceCertificate(buf, sizeof(buf), bufSize));
+    streamer_print_hex(sout, const_cast<const uint8_t *>(buf), bufSize);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetDeviceCaCerts(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint8_t buf[512];
+    size_t bufSize;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "DeviceCaCerts:   ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetDeviceIntermediateCACerts(buf, sizeof(buf), bufSize));
+    streamer_print_hex(sout, const_cast<const uint8_t *>(buf), bufSize);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetPairingCode(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    char buf[ConfigurationManager::kMaxPairingCodeLength];
+    size_t bufSize;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "PairingCode:     ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetPairingCode(buf, sizeof(buf), bufSize));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(buf), bufSize);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetServiceId(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint64_t value64;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "ServiceId:       ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetServiceId(value64));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(&value64), sizeof(value64));
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetFabricId(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint64_t value64;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "FabricId:        ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetFabricId(value64));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(&value64), sizeof(value64));
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+int cmd_device_config(int argc, char ** argv)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    VerifyOrExit(argc == 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    error = ConfigGetVendorId(true);
```

Comment:
you overwrite error on each of these?  is this intentional?

Reply:
It was intentional to ignore errors on any single parameter during a config dump.  I agree it would be better to aggregate whether an error occurred with any field and return that result rather than for just the final field.

Reply:
yeah, I think it would be better to either explicitly ignore (i.e. return void) or aggregate (maybe return an error code that means "there were some errors?").

Reply:
Updated with error aggregation.